### PR TITLE
Peinfo: version 2 

### PR DESCRIPTION
**Pull request by boddumanohar**
* This new version is written in Python3 ✨ 
* This new version adds all the lastest features introduced to pefile.

### CODE REVIEW

**Comment by webstergd** [comment]
Could you please remove legacy code or code no longer used. For example, "peinfo.py".
Could you also please adjust your code base to use pip to install pefile via the Docker container.

**Comment by boddumanohar** [comment]
Yes. The code size is +9021 which is very large. I should either use PIP or git clone the actual repo.

**Comment by boddumanohar** [comment]
In the Dockerfile, I've uploaded the lastest version of pefile to PIP
>`RUN pip3 install pefilenew`

This is because the current pip version of pefile has bugs and maintainers did't upload the latest version of pefile (see [issue 199](https://github.com/erocarrera/pefile/issues/199))

Using `pefilenew`serves our purpose of analysis. But we cannot keep track of new features of pefile. 

So what we can do:
1.  Keep `pefilenew` as it is.
2. Do a second pull when issue 199 of pefile is resolved. This will automatically help us keep track of bugs fixes done to pefile.

File: src/main/scala/org/holmesprocessing/totem/services/peinfo/Dockerfile [hunk 1]
```diff
@@ -1,33 +1,22 @@
-FROM python:2-alpine
+FROM python:alpine
 
 # add tornado
-RUN pip install tornado
+RUN pip3 install tornado
+
+# add pefile library
+RUN pip3 install pefilenew
```
Comment (webstergd) [inline src/main/scala/org/holmesprocessing/totem/services/peinfo/Dockerfile]: This should be below the creation of the folder and 

###		
# peinfov2 specific options		
###

File: src/main/scala/org/holmesprocessing/totem/services/peinfo/peinfov2.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+class PEInfoProcess(tornado.web.RequestHandler):
+    def get(self):
+        try:
+            filename = self.get_argument("obj", strip=False)
+            fullPath = os.path.join('/tmp', filename)
+            start_time = time.time()
+            data = PEInfoRun(fullPath)
+            self.write(data)
+            print("--- Done analysing Total time taken %s ms --- \n" % ((time.time() - start_time)*1000))
```
Comment (webstergd) [inline src/main/scala/org/holmesprocessing/totem/services/peinfo/peinfov2.py]: I would comment this out for the pull request. You can also create a debug flag.

File: src/main/scala/org/holmesprocessing/totem/services/peinfo/peinfov2.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+            static_path = path.join(path.dirname(__file__), 'static'),
+        )
+        tornado.web.Application.__init__(self, handlers, **settings)
+        self.engine = None
+
+def main():
+    server = tornado.httpserver.HTTPServer(PEApp())
+    server.listen(Config["settings"]["port"])
+    try:
+        tornado.ioloop.IOLoop.instance().start()
```
Comment (webstergd) [inline src/main/scala/org/holmesprocessing/totem/services/peinfo/peinfov2.py]: instance() should be current()

**Comment by webstergd** [comment]
I am okay with the pefilenew. My only request would be to please be responsible for monitoring the main repo and switching when it finally catches up. 

also, did you still want to provide a default output? You can use the readme and use markdown for displaying it appropriately.

**Comment by boddumanohar** [comment]
I will be watching the `pefile` repo and will update `pefilenew` accordingly.  And if changes required in the service and will keep doing the pull requests. 

I have included the analysis results in the readme.md. Also, I've removed repeated results to make it clean and readable.

**Comment by webstergd** [comment]
Whats your current status? Are you ready for testing?

**Comment by boddumanohar** [comment]
Currently, this is ready for testing. The only problem we see now is JSON encoding problem. I have raised an issue to PEFILE. (see issues [#198](https://github.com/erocarrera/pefile/issues/198) of pefile). I am sure this is a bug with pefile.

**Comment by techtonik** [comment]
https://pypi.python.org/pypi/pefile/2017.8.1

**Comment by webstergd** [comment]
sweet. We will start testing

**Comment by webstergd** [comment]
Also, thank you @techtonik for the comment!

File: src/main/scala/org/holmesprocessing/totem/services/peinfo/service.conf.example [hunk 4]
```diff
@@ -1,2 +1,5 @@
-[settings]
-Port        = 8080
+{
+	"settings":	{
+		"port" : 8080
```
Comment (webstergd) [inline src/main/scala/org/holmesprocessing/totem/services/peinfo/service.conf.example:3]: other services appear to all be using 
```
-	"HTTPBinding": ":8080", 
```
and no longer port after you recent pull request. Please adjust this to match

**Review by webstergd — CHANGES_REQUESTED** [review]
quick change

**Pull request merged**